interface/MuonsProducer.h

```diff
@@ -47,7 +68,10 @@ class MuonsProducer: public LeptonsProducer<pat::Muon>, public ScaleFactors {
         BRANCH(dxy, std::vector<float>);
         BRANCH(dz, std::vector<float>);
         BRANCH(dca, std::vector<float>);
-        rochcor2015 rmcor;
+        std::unique_ptr<rochcor2016> rmcor;
+        std::unique_ptr<KalmanMuonCalibrator> kamucacor;
+        BRANCH(rochester_correction, std::vector<float>);
+        BRANCH(kamuca_correction, std::vector<float>);
```

Comment:
One branch is enough since you explicitly check if both method are used at the same time. Something like `scale_correction_factor`?